bin/launch_check_model.py

```diff
@@ -49,39 +51,85 @@ def check_model(self):
 
         x, y, meta = next(iter(self.train_data))
         # train the model for one epoch
-        loss, output = self.model.batch(x, y, self.loss_fn, self.optimizer)
+        loss, output = self.model.batch(x, y, self.loss_dict, self.optimizer)
         # check the model weights have changed, and print if it has
         for key in initial_model_weights:
             if torch.equal(initial_model_weights[key], self.model.state_dict()[key]):
                 print(f"Model weights have not changed for key {key}")
             else:
                 print(f"Model weights have changed for key {key}")
 
-        # print the computed loss, displaying loss_fn name as well
-        print(f"Loss computed with {self.loss_fn.__class__.__name__} is {loss.item()}")
-
+        # print the computed loss, displaying loss_dict as well
+        print(f"Loss computed with {self.loss_dict} is {loss.item()}")
+"""
 
-def main(data: str, model_file: str, experiment_config: str, config_file: str):
+def main(data_path: str, model_path: str, experiment_config: str, config_path: str, num_samples: int):
 
     # TODO update to yaml the experimnt config
     # load json into dictionary
     exp_config = {}
     with open(experiment_config, 'r') as in_json:
         exp_config = json.load(in_json)
-    
+
     # Initialize json schema it checks for correctness of the Json architecture and fields / values. already raises errors.
     schema = JsonSchema(exp_config)
 
-    Model = import_class_from_file(model_file)
-    experiment = get_experiment(schema.experiment)
+    # initialize the experiment class
+    initialized_experiment_class = get_experiment(schema.experiment)
+
+    # import the model correctly but do not initialize it yet, ray_tune does that itself
+    model_class = import_class_from_file(model_path)
+
+    # Update the tune config file. no need to run the whole amount asked for the tuning. Basically downsample the tuning.
+    updated_tune_conf = "check_model_modified_tune_config.yaml"
+    with open(config_path, 'r') as conf_file, open(updated_tune_conf, "w") as new_conf:
+        user_tune_config = yaml.safe_load(conf_file)
+        # make so the tune run just once per num_sample
+        user_tune_config["tune"]["tune_params"]["num_samples"]          = num_samples
```

Comment:
so this config is only changed for checking purpose?
The config file for the real tuning in the pipeline will not be overwritten?

Reply:
exactly.